src/common/uuid.cpp

```
@@ -10,6 +10,45 @@
 
 namespace Common {
 
+namespace {
+
+constexpr u8 HexCharToByte(char c) {
+    if (c >= '0' && c <= '9') {
+        return c - '0';
+    }
+    if (c >= 'a' && c <= 'f') {
+        return c - 'a' + 10;
+    }
+    if (c >= 'A' && c <= 'F') {
+        return c - 'A' + 10;
+    }
+    return 0;
+}
+
+} // Anonymous namespace
+
+constexpr u128 UUID::HexStringToU128(std::string_view hex_string) const {
```

Shouldn't this function assert if a given string has enough bytes in it that they exceed the 128-bit range?

---

I don't think silently discarding characters in the string is a good idea, because it can hide genuine logic bugs.